### DISCUSSION TITLE
Type 'never' has no call signatures even if Atom initilized with value.

### DISCUSSION THREAD

**Discussion by MarianiGiacomo** [pmndrs/jotai#1232] [Q&A]
I get the TypeScript warning `This expression is not callable.
  Type 'never' has no call signatures.` even if the atom is initialized with a value. It infers type `never` for `setQuantity` (`const setQuantity: never`)
```
export const cartQuantityAtom = atom<number>(0)
```
 ```
const [quantity, setQuantity] = useAtom<number>(cartQuantityAtom)
```
```
setQuantity(0)
```
`tsconfig.json`: 
```
{
  "compilerOptions": {
    "noEmit": true,
    "target": "esnext",
    "module": "commonjs",
    "declaration": true,
    "outDir": "lib",
    "strict": true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "skipLibCheck": true,
    "jsx": "preserve",
  },
  "exclude": ["node_modules", "public", ".cache"]
}
```
I have looked at #802, #550, and #352 but I still can't figure out how to fix it.

**Comment by dai-shi** [comment]
Can you try this?
```js
const [quantity, setQuantity] = useAtom(cartQuantityAtom)
```

**Reply by MarianiGiacomo** [reply]
That way I don't get any warnings. So I guess the problem was my misunderstanding on how to use TypeScript in `useAtom`, and it is not necessary to specify the type since it is already specified when declaring the atom?

**Reply by dai-shi** [reply]
Yeah, we recommend to infer types with `useAtom`.
